edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/storage/MessageStore.cs

```diff
@@ -536,6 +536,8 @@ public MessageWithOffset(IMessage message, long offset)
 
             public long Offset { get; }
 
+            public uint ProcessedPriority { get; set; }
```

Comment:
Hmm, is this a breaking change? All current messages will be Pri 0, as opposed to default, right?
And don't make the property settable

Reply:
I'm assuming you couldn't tell how this was used because StoringAsyncEndpointExecutor.cs got CRLF stomped.

But the set is necessary because the MessageStore doesn't understand priority when it re-constructs the message objects queue.  So the executor layer needs a way to tag the messages with their priority after the fact.

So this new property is only used for plumbing priority info for the purpose of metrics, and has no impact on any business logic.

---
In reply to: [397348395](https://github.com/Azure/iotedge/pull/2730#discussion_r397348395) [](ancestors = 397348395)

Reply:
We keep all objects as immutable.. this breaks that pattern. Notice the same pattern with this object. We should keep the same pattern.

Reply:
Also, why can't we store the message priority and then load it as well?